pkg/api/component_report_test.go

```diff
@@ -0,0 +1,1183 @@
+// nolint
+package api
+
+import (
+	apitype "github.com/openshift/sippy/pkg/apis/api"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func fakeComponentAndCapabilityGetter(name string) (string, []string) {
+	known := map[string]struct {
+		component    string
+		capabilities []string
+	}{
+		"test 1": {
+			component:    "component 1",
+			capabilities: []string{"cap1"},
+		},
+		"test 2": {
+			component:    "component 2",
+			capabilities: []string{"cap21", "cap22"},
+		},
+	}
+	if comCap, ok := known[name]; ok {
+		return comCap.component, comCap.capabilities
+	}
+	return "other", []string{"other"}
+}
+
+func TestGenerateComponentReport(t *testing.T) {
```

Comment:
Looks like good test coverage, but I feel like a 1200-line test function is going to be hard to maintain. Is it possible to make it's footprint smaller?

Some of the test cases look reused repeatedly -- maybe you could use some file scope consts/vars like this:


```golang
var awsAMD64OVNTest = apitype.ComponentTestIdentification {
					TestName: "test 1",
					TestID:   "1",
					Platform: "aws",
					Arch:     "amd64",
					Network:  "ovn",
}
```

Likewise maybe for the results, you can have some consts for these: 

```golang
{
    TotalCount:   1000,
    FlakeCount:   10,
    SuccessCount: 900,
}
```

And maybe it might sense to break it up into multiple functions.